strawberryfields/apps/similarity.py

```diff
@@ -412,119 +527,218 @@ def prob_event_mc(
+        raise ValueError("List of orbits should only consist of orbits")
+    if any(min(item) < 0 for item in list_of_orbits):
+        raise ValueError("Cannot request orbits with photon number below zero")
+    if n_mean < 0:
+        raise ValueError("Mean photon number must be non-negative")
+    if not 0 <= loss <= 1:
+        raise ValueError("Loss parameter must take a value between zero and one")
+    if mc_samples:
+        return [_prob_orbit_mc(graph, orbit, n_mean, mc_samples, loss) for orbit in list_of_orbits]
 
-    return [count[p] / n_samples for p in event_photon_numbers]
+    return [_prob_orbit_exact(graph, orbit, n_mean, loss) for orbit in list_of_orbits]
 
 
-def feature_vector_mc(
+def feature_vector_events(
     graph: nx.Graph,
     event_photon_numbers: list,
     max_count_per_mode: int = 2,
     n_mean: float = 5,
-    samples: int = 1000,
+    mc_samples: int = None,
     loss: float = 0.0,
 ) -> list:
-    r"""Calculates feature vector using Monte Carlo estimation of event probabilities according to the
-    input graph.
+    r"""Calculates feature vector made up of either exact or MC-approximated event probabilities for the input graph.
 
-    The feature vector is composed of event probabilities with a fixed maximum photon count in
-    each mode but a range of total photon numbers specified by ``event_photon_numbers``.
+    These probabilities can be either exact or estimated using Monte Carlo estimation.
+    The argument ``mc_samples`` is set to ``None`` to get exact feature vector by default.
+    To use Monte Carlo estimation, ``mc_samples`` can be set to the number of samples desired
+    to be used in the estimation.
 
-    Probabilities are reconstructed using Monte Carlo estimation.
+    Computing exact probabilities for a large number of events especially for events with high
+    total photon numbers can be quite time-consuming. For example, calculating the exact
+    probabilities of observing 8 total photons in a 25 mode graph can take on the order of a
+    few minutes. This means that if the dataset has 1000 graphs, obtaining feature vectors for
+    all can take many days. Monte Carlo estimation, although less precise, can be much quicker.
 
     **Example usage:**
 
     >>> graph = nx.complete_graph(8)
-    >>> feature_vector_mc(graph, [2, 4, 6], 2)
-    [0.2115, 0.1457, 0.09085]
+    >>> feature_vector_events(graph, [2,3,4,6], 1)
+    [0.18423136914879445, 0.0, 0.07020748298295838, 0.011891063791931982]
+
+    >>> feature_vector_events(graph, [2,3,4,6], 1, mc_samples = 1000)
+    [0.18423136914879246, 0.0, 0.07020748298295802, 0.011891063791932176]
 
     Args:
         graph (nx.Graph): input graph
-        event_photon_numbers (list[int]): a list of events described by their total photon number
-        max_count_per_mode (int): maximum number of photons per mode for all events
+        event_photon_number (int): number of photons in the event

```

Comment:
Why is this line changed from before? I.e.., shouldn't it be `event_photon_numbers (list[int]): a list of events described by their total photon number`